packages/scheduler/src/clock.js

```diff
@@ -0,0 +1,48 @@
+/** @license MIT License (c) copyright 2010-2017 original author or authors */
+
+/*global performance, process*/
+
+export class MillisecondClock {
+  constructor (now, origin) {
+    this.origin = origin
+    this._now = now
+  }
+
+  now () {
+    return this._now() - this.origin
+  }
+}
+
+export class HRTimeClock {
+  constructor (hrtime, origin) {
+    this.origin = origin
+    this._hrtime = hrtime
+  }
+
+  now () {
+    const hrt = this._hrtime(this.origin)
+    return (hrt[0] * 1e9 + hrt[1]) / 1e6
+  }
+}
+
+export const millisecondClockFromNow = now =>
+  new MillisecondClock(now, now())
+
+export const newPerformanceNowClock = () =>
+  millisecondClockFromNow(performance.now)
```

Comment:
This should be `performance.now.bind(performance)`. It's failing for me on Firefox.